tests/test_dirty_local.py

```diff
@@ -4,7 +4,7 @@
 import pytest
 from plumbum import local
 from plumbum.cmd import git
-from pytest_gitconfig import GitConfig
+from pytest_gitconfig.plugin import GitConfig
```

Comment:
Did mypy complain? The [docs](https://pypi.org/project/pytest-gitconfig/) show that we should import from `pytest_gitconfig` directly, and the [code](https://github.com/noirbizarre/pytest-gitconfig/blob/main/src/pytest_gitconfig/__init__.py) correctly imports `GitConfig` in the top-level module.

By the way the docs show that this import should be guarded under `if TYPE_CHECKING`, does it apply to us here as well? Should we `from __future__ import annotations` in all tests module (that's what I do in my own projects)?

Reply:
> Did mypy complain? The [docs](https://pypi.org/project/pytest-gitconfig/) show that we should import from `pytest_gitconfig` directly, and the [code](https://github.com/noirbizarre/pytest-gitconfig/blob/main/src/pytest_gitconfig/__init__.py) correctly imports `GitConfig` in the top-level module.

mypy complains because `GitConfig` is not explicitly re-exported from `pytest_gitconfig`. That could be fixed with an upstream change


> By the way the docs show that this import should be guarded under `if TYPE_CHECKING`, does it apply to us here as well?

not required, but something i could add in a future PR

> Should we `from __future__ import annotations` in all tests module (that's what I do in my own projects)?

also not required, since ruff can detect when it is needed and automatically add it